worlds/dark_souls_3/__init__.py

```diff
@@ -1457,19 +1458,20 @@ def smooth_items(item_order: List[Union[DS3ItemData, DarkSouls3Item]]) -> None:
                         f"contain smoothed items, but only {len(converted_item_order)} items to smooth."
                     )
 
+                sorted_spheres = []
                 for sphere in locations_by_sphere:
                     locations = [loc for loc in sphere if loc.item.name in names]
 
                     # Check the game, not the player, because we know how to sort within regions for DS3
                     offworld = ds3_world._shuffle([loc for loc in locations if loc.game != "Dark Souls III"])
                     onworld = sorted((loc for loc in locations if loc.game == "Dark Souls III"),
                                      key=lambda loc: loc.data.region_value)
+                    sorted_spheres.extend(onworld)
+                    sorted_spheres.extend(offworld)
 
-                    # Give offworld regions the last (best) items within a given sphere
-                    for location in onworld + offworld:
-                        new_item = ds3_world._pop_item(location, converted_item_order)
-                        location.item = new_item
-                        new_item.location = location
+                # Give offworld regions the last (best) items within a given sphere
+                converted_item_order.reverse()
+                remaining_fill(multiworld, sorted_spheres, converted_item_order, name="DS3 Smoothing", check_location_can_fill=True)
```

Comment:
It sounds like your testing indicates that this currently respects the ordering of `locations`, but is that a guaranteed part of the contract that won't change moving forward? It's hard to tell since the function itself isn't documented. I want to avoid a situation where it gets refactored in the future and suddenly item smoothing stops working.

Reply:
I would be extremely shocked if this ever changed.

Reply:
Would it be possible to add a brief doc comment to `remaining_fill` to indicate that this is an intended part of the behavior?

Reply:
@NewSoupVi thoughts?

Reply:
Uhhh I mean I agree with this PR because I specifically added this functionality to remaining_fill so that DS3 can do this, but I guess I hadn't thought about whether this would always be guaranteed

All our fill functions currently go in order so it seems intentional, and I would hope that changing that would trip up whichever core maintainer is looking at it

I guess we could add a comment tho to really make sure